tests/Functional/SelfCommandsTest.php

```diff
@@ -44,6 +43,14 @@ public function testSelfUpdateCommand()
             'Failed installing plugins to setup self:update command test.'
         );
         $output = $this->terminus(self::SELF_UPDATE_COMMAND);
+        [$output, $exitCode, $error] = static::callTerminus(self::SELF_UPDATE_COMMAND);
+        if (
+            0 !== $exitCode
+            && false !== strpos($error, 'rate limit exceeded')
+        ) {
+            $this->markTestSkipped(sprintf('Failed executing %s command: %s', self::SELF_UPDATE_COMMAND, $error));
```

Comment:
Looks like on lines 51 and 28 we're skipping the tests if github rate limits us.  That seems reasonable.

~By my reading, these lines will only execute in the rate limiting scenario so let's clarify in the skipped message that rate limiting is specifically what happened.~  
I guess the inclusion of the error message itself makes that clear "enough".  I'm probably going to suggest an adjustment, though.